packages/Chem/src/rdkit-service/rdkit-service-worker-similarity.ts

```diff
@@ -17,43 +18,91 @@ export class RdKitServiceWorkerSimilarity extends RdKitServiceWorkerBase {
     const fps: (Uint8Array | null)[] = [];
     try {
       switch (fingerprintType) {
-      case Fingerprint.Pattern:
-        for (let i = 0; i < this._rdKitMols.length; ++i) {
-          try {
-            if (!this._rdKitMols[i])
+        case Fingerprint.Pattern:
+          for (let i = 0; i < this._rdKitMols.length; ++i) {
+            try {
+              if (!this._rdKitMols[i])
+                fps.push(null);
+              else
+                fps.push(this._rdKitMols[i]!.get_pattern_fp_as_uint8array());
+            } catch {
               fps.push(null);
-            else
-              fps.push(this._rdKitMols[i]!.get_pattern_fp_as_uint8array());
-          } catch {
-            fps.push(null);
-            continue;
+              continue;
+            }
           }
-        }
-        break;
-      case Fingerprint.Morgan:
-        for (let i = 0; i < this._rdKitMols.length; ++i) {
-          try{
-            if (!this._rdKitMols[i] || this._rdKitMols[i] && this._rdKitMols[i]?.is_qmol)
+          break;
+        case Fingerprint.Morgan:
+          for (let i = 0; i < this._rdKitMols.length; ++i) {
+            try {
+              if (!this._rdKitMols[i] || this._rdKitMols[i] && this._rdKitMols[i]?.is_qmol)
+                fps.push(null);
+              else
+                fps.push(this._rdKitMols[i]!.get_morgan_fp_as_uint8array(JSON.stringify({
+                  radius: this._fpRadius,
+                  nBits: this._fpLength,
+                })));
+            } catch (error) {
               fps.push(null);
-            else
-              fps.push(this._rdKitMols[i]!.get_morgan_fp_as_uint8array(JSON.stringify({
-              radius: this._fpRadius,
-              nBits: this._fpLength,
-            })));
-          } catch (error) {
-            fps.push(null);
-            continue;
+              continue;
+            }
           }
-        }
-        break;
-      default:
-        throw Error('Unknown fingerprint type: ' + fingerprintType);
+          break;
+        default:
+          throw Error('Unknown fingerprint type: ' + fingerprintType);
       }
     } catch (e) {
       // nothing to do, bit is already 0
     }
     return fps!.map((el: any) => {
-      return {data: el, length: el ? el.length : 0};
+      return { data: el, length: el ? el.length : 0 };
+    });
+  }
+
+
+  getFingerprintsWithMolsOnFly(dict: string[], fingerprintType: Fingerprint) {
+    const fps: (Uint8Array | null)[] = [];
+    for (let i = 0; i < dict.length; ++i) {
+      const item = dict[i];
+      let mol;
+      if (!item || item === '')
+        fps.push(new Uint8Array());
```

Comment:
Because for empty molecules we want to get an empty bitset insted of null fingerprint (null is used for malformed data)

Reply:
Why would we want an empty bitset? We want to exclude it from all further computations, there is literally no molecule, end of story :)

Reply:
Since we add a column with fingerprints to an initial dataframe, number of rows in returned fingerprints column should match the initial dataframe. So we need to keep rows for empty/malformed molecules. After, of course, we do not include those malformed/empty data into analisys.
And since we want to separate empty molecules from malfomed molecules (to warn users only of malformed molecules indexes) we separate them - empty bitsets for empty, nulls - for malformed. 
Btw, not only undefined or '' matches empty molecules. We can also have empty molv2000 or molv3000 which is basically not an empty string. And in case of empty molfile rdkit itself creates empty fingerprint bitset.